client/src/main/java/org/apache/rocketmq/schema/registry/client/serde/avro/AvroDeserializer.java

```diff
@@ -59,7 +58,11 @@ public void configure(Map<String, Object> configs) {
     }
 
     @Override
-    public T deserialize(String subject, byte[] payload)
+    public T deserialize(String subject, byte[] payload) {
+        return this.deserialize(subject, payload, null);
+    }
```

Comment:
Can this method support both bySubject and byRecordId?  For example, add a configuration item for _SerdeConfig_?
Because byRecordId is applicable to message and rocketmq-connect, but not to rocketmq-streams, because the streams scenario cannot require that the data written to MQ must have a recordId, which will greatly reduce the flexibility of streams applications.

能否通过这一个方法，同时支持bySubject和 byRecordId 两种方式？？ 比如增加 SerdeConfig 的一个配置项？
因为byRecordId 适用于message场景和connect场景，但是不适用于streams场景，streams场景无法要求写入MQ的数据一定要带着recordId，这会大大降低streams应用的灵活性。

Reply:
不太理解 Streams 场景为何不能要求数据一定带着 recordId，本质上增加 schema 就是通过限制来保证链路上的类型安全，如果为了灵活性去掉这个限制的话，感觉和旧版本没有太大的区别了

但是从实现角度上增加一种不依赖 Schema server 的分支，我觉得也没问题 @humkum 可以考虑先加上

Reply:
1、byRecordId 更适合于message场景和connect场景，这两个场景下注重“限制”、“类型安全”，我觉得没问题。

2、但是schema的作用不会仅限于“限制”和“数据治理”，我认为另一个大的价值是“Table化”和“列读”，尤其在ETL 场景。假设我们原来有一个topic，Producer已经生产了大量的存量消息，我们如果要用streams分析这些消息数据，会因为没有RecordId 直接无法解析而浪费掉。

3、我想表达的也不全是 “增加一种不依赖 Schema server 的分支”，而是我们的消息解析 可以同时支持 deserialize bySubject 和 byRecordId ，这两种都是依赖registry的，都是能保障类型安全的，只不过bySubject 的保障稍微弱一些，byRecordId 严格以版本为保障，类型安全更强。 但是 bySubject 可以让消息不强依赖 recordId，我们能分析的RocketMQ数据源才能更多、更灵活。

Reply:
刚和坤明沟通了下，之前误解了 bySubject 的含义，现在这个理解应该是一致的 @MatrixHB